### PR TITLE
feat(discover): Reset Discover page when navigated to

### PR DESCRIPTION
**Pull request by lynnagara**
When Discover is navigated to again, reset the page

### CODE REVIEW

File: tests/js/spec/views/organizationDiscover/discover.spec.jsx [hunk 1]
```diff
@@ -116,12 +117,24 @@ describe('Discover', function() {
   describe('reset()', function() {
     let wrapper, queryBuilder;
     beforeEach(function() {
+      browserHistory.push = function(url) {
+        wrapper.setProps({
```
Comment (billyvg) [inline tests/js/spec/views/organizationDiscover/discover.spec.jsx]: I think by default `browserHistory.push` is a jest mock fn, so you could use `browserHistory.push.mockImplementationOnce(fn)`.

I think what would be helpful to test here is: click Reset button, assert that `push` gets called, assert `queryBuilder.reset()` gets called, and then we should also check that state gets reset?

File: src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx [hunk 2]
```diff
@@ -44,6 +44,18 @@ export default class OrganizationDiscover extends React.Component {
     };
   }
 
+  componentWillReceiveProps(nextProps) {
+    const {queryBuilder, location: {search}} = nextProps;
+    if (search === '') {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx]: Is it possible that `search` here be any other falsey string? Can we early return here?

File: tests/js/spec/views/organizationDiscover/discover.spec.jsx (hunk 1 above)
Comment (lynnagara) [reply tests/js/spec/views/organizationDiscover/discover.spec.jsx]: Added expectations for state variables and `queryBuilder.reset()`

File: tests/js/spec/views/organizationDiscover/discover.spec.jsx [hunk 3]
```diff
@@ -133,6 +147,16 @@ describe('Discover', function() {
       wrapper.update();
     });
 
+    it('resets query builder and state', function() {
+      wrapper.instance().reset();
```
Comment (billyvg) [inline tests/js/spec/views/organizationDiscover/discover.spec.jsx]: I think there's some value in testing the contract of the the Reset button itself with the reset handler.

**Review by billyvg — APPROVED** [review]

File: tests/js/spec/views/organizationDiscover/discover.spec.jsx (hunk 3 above)
Comment (lynnagara) [reply tests/js/spec/views/organizationDiscover/discover.spec.jsx]: Good point, so far have just tested functions in discover, not any jsx etc but will keep this in mind

**Pull request merged**